src/g.cpp

```diff
@@ -1164,10 +1169,10 @@ g2 g2::frobeniusMap(int64_t power) const
     return r;
 }
 
-// MultiExp calculates multi exponentiation. Given pairs of G2 point and scalar values
+// WeightedSum calculates multi exponentiation. Given pairs of G2 point and scalar values
 // (P_0, e_0), (P_1, e_1), ... (P_n, e_n) calculates r = e_0 * P_0 + e_1 * P_1 + ... + e_n * P_n
 // Length of points and scalars are expected to be equal, otherwise NONE is returned.
-optional<g2> g2::multiExp(const vector<g2>& points, const vector<array<uint64_t, 4>>& scalars, std::function<void()> yield)
+optional<g2> g2::weightedSum(const vector<g2>& points, const vector<array<uint64_t, 4>>& scalars, std::function<void()> yield)
```

Comment:
Make `yield` a `const&`. Same for `g1` version. Perf issue as yield is unnecessarily copied for each invocation.

Reply:
fixed